ngi_pipeline/tests/engines/sarek/models/test_sarek.py

```diff
@@ -102,29 +104,86 @@ def _alignment_status_wrapper(status, *args):
 
     def test_configure_analysis(self, charon_connector_mock, reference_genome_mock):
         config = {
-            "profile": "this-is-a-profile",
-            "tools": ["tool-A", "tool-B"]}
+            "nextflow": {
+                "config": "this-is-a-site-specific-config-file",
+                "profile": "this-is-another-profile",
+                "nf_path": "this-is-the-path-to-nextflow"},
+            "sarek": {
+                "sarek_path": "this-is-the-path-to-sarek",
+                "unique-key": "this-is-not-in-the-default-config",
+                "tools": ["tool-A", "tool-B"]}}
 
         sarek_analysis = SarekAnalysis(
             reference_genome_mock.return_value,
-            {"sarek": config},
+            config,
             self.log,
             charon_connector=charon_connector_mock.return_value)
 
-        for key in config.keys():
-            self.assertEqual(config[key], sarek_analysis.sarek_config[key])
-        self.assertEqual(SarekAnalysis.DEFAULT_CONFIG["nf_path"], sarek_analysis.sarek_config["nf_path"])
-        self.assertEqual(SarekAnalysis.DEFAULT_CONFIG["sarek_path"], sarek_analysis.sarek_config["sarek_path"])
+        for key in config["sarek"].keys():
+            self.assertEqual(config["sarek"][key], sarek_analysis.sarek_config[key])
+        for key in config["nextflow"].keys():
+            self.assertEqual(config["nextflow"][key], sarek_analysis.nextflow_config[key])
+
+        del(config["nextflow"]["profile"])
```

Comment:
This test is really long and a bit hard to follow, I think it could benefit from being divided into different tests. For example this part when testing if the default value is used.

Reply:
good point, I'll look into that